src/middleware/validate.js

```diff
@@ -0,0 +1,17 @@
+export function validateGroupId(req, res, next) {
+  const groupId = parseInt(req.params.groupId, 10);
+
+  if (isNaN(groupId)) {
+    return res.status(400).json({ message: 'groupId must be integer' });
+  }
+  next();
```

Comment:
```suggestion
  req.groupId = groupId
  next();
```
여기서 그룹 아이디 유효성 검증을 하는데 컨트롤러에서 한 번 더 파싱을 하는 것 같습니다. 미들웨어에서 파싱한 값을 `req` 객체에 담아 컨트롤러로 전달 하는 것도 좋을 것 같습니다!